app/controllers/submissions_controller.rb

```diff
@@ -14,26 +14,22 @@ def index
   end
 
   def create
-    @post = Submission.new(post_params)
-    @post.user_id = current_user.id
-    if @post.save
-      flash[:alert] = "Your submission was saved"
-      redirect_to submission_path(@post.id)
+    if !params[:submission][:post_id].present?
+      @submission = current_user.submissions.build(post_params)
+      if @submission.save
+        flash[:alert] = "Your submission was saved"
+        redirect_to submission_path(@submission.id)
+      else
+        render :new
+      end
     else
-      render :new
-    end
-  end
-
-  def create_comment
-    @comment = Submission.new(comment_params)
-    @comment.user_id = current_user.id
-    @comment.post_id = parent_post_id
-    if @comment.save
-      flash.now[:alert] = "Your submission was saved"
-      redirect_to submission_path(@comment.post_id)
-    else
-      flash[:alert] = @comment.errors.messages
-      redirect_to submission_path(parent_post_id)
+      @comment = current_user.submissions.build(comment_params)
+      if @comment.save
+        redirect_to submission_path(@comment.post_id)
+      else
+        flash[:inline] = @comment.errors.full_messages_for(:text).join(", ")
+        redirect_to submission_path(@comment.post_id)
+      end
```

Comment:
This conditional can be avoided by adding a `CommentsController` and moving this comment creation logic to the `create` action there.

Reply:
I'm wondering if it would be beneficial to also have a `new` view to handle rendering errors...